controllers/toolchaincluster/toolchaincluster_controller_test.go

```diff
@@ -147,6 +207,21 @@ func newToolchainCluster(name, tcNs string, apiEndpoint string, status toolchain
 }
 
 func prepareReconcile(toolchainCluster *toolchainv1alpha1.ToolchainCluster, cl *test.FakeClient, requeAfter time.Duration) (Reconciler, reconcile.Request) {
+	controller := Reconciler{
+		Client:     cl,
+		Scheme:     scheme.Scheme,
+		RequeAfter: requeAfter,
+		CheckHealth: func(context.Context, *kubeclientset.Clientset) []toolchainv1alpha1.Condition {
+			return toolchainCluster.Status.Conditions
+		},
```

Comment:
This is a bit misleading to me. I read it as "return the conditions the cluster already has".

I'd suggest to remove the `prepareCheckHealthDefaultReconcile` function and to update the `prepareReconcile` function to require the function to use as `CheckHealth`. This will make harder to inadvertently test the healthchecker logic from this suite.

WDYT?

Reply:
This is to check second the scenario in reconciler, where we would pass the conditions when we call the reconciler. i wouldnt wont to remove this, probably i can update the return conditions

Reply:
i think i have updated it now